### PR TITLE
Quickfixs

### PR DESCRIPTION
**Pull request by Psycojoker**
Hello,

A serie of quickfix I've made while reading this file (I wanted to know how domains were stored).

You can cherry-pick if you want.

### CODE REVIEW

**Comment by Psycojoker** [comment]
I still need to make some tests, don't merge too quickly :$

File: src/yunohost/domain.py [hunk 1]
```diff
@@ -200,12 +194,13 @@ def domain_remove(auth, domain, force=False):
                                           m18n.n('domain_uninstall_app_first'))
 
     if auth.remove('virtualdomain=' + domain + ',ou=domains') or force:
-        os.system('rm -rf /etc/yunohost/certs/%s' % domain)
+        shutil.rmtree('rm -rf /etc/yunohost/certs/%s' % domain)
```
Comment (jeromelebleu) [inline src/yunohost/domain.py]: Oops, it seems there is an issue here! :)

File: src/yunohost/domain.py [hunk 2]
```diff
@@ -160,12 +154,14 @@ def domain_add(auth, domain, dyndns=False):
             with open('/etc/yunohost/installed', 'r') as f:
                 service_regen_conf(names=[
                     'nginx', 'metronome', 'dnsmasq', 'rmilter'])
-                os.system('yunohost app ssowatconf > /dev/null 2>&1')
+                assert os.system('yunohost app ssowatconf > /dev/null 2>&1') == 0, "SSOwat conf regen failed"
```
Comment (jeromelebleu) [inline src/yunohost/domain.py]: It's really ugly and should be replaced by `app_ssowatconf` method call directly......

File: src/yunohost/domain.py [hunk 3]
```diff
@@ -200,12 +194,13 @@ def domain_remove(auth, domain, force=False):
...
 
     if auth.remove('virtualdomain=' + domain + ',ou=domains') or force:
-        os.system('rm -rf /etc/yunohost/certs/%s' % domain)
+        shutil.rmtree('rm -rf /etc/yunohost/certs/%s' % domain)
     else:
         raise MoulinetteError(errno.EIO, m18n.n('domain_deletion_failed'))
 
     service_regen_conf(names=['nginx', 'metronome', 'dnsmasq'])
     os.system('yunohost app ssowatconf > /dev/null 2>&1')
+    assert os.system('yunohost app ssowatconf > /dev/null 2>&1') == 0, "SSOwat conf regen failed"
```
Comment (jeromelebleu) [inline src/yunohost/domain.py]: Same here, it should call the `app_ssowatconf` method!

**Comment by jeromelebleu** [comment]
Thank you for this really big review!! :)

**Comment by zamentur** [comment]
What is the status of this PR ?

**Comment by Psycojoker** [comment]
Probably too big and outdated now, I'm closing it and I'll redo one or a series of smaller one (or directly push on unstable if it's only things like pep8 which are just looks and not possible breaking things at a time where we don't have too much PRs).

**Pull request closed without merging**